src/sentry/backup/helpers.py

```diff
@@ -27,6 +34,110 @@ def default(self, obj):
         return super().default(obj)
 
 
+def create_encrypted_export_tarball(
+    json_export: json.JSONData, encrypt_with: BinaryIO
+) -> io.BytesIO:
+    """
+    Generate a tarball with 3 files:
+
+      1. The DEK we minted, name "data.key".
+      2. The public key we used to encrypt that DEK, named "key.pub".
+      3. The exported JSON data, encrypted with that DEK, named "export.json".
+
+    The upshot: to decrypt the exported JSON data, you need the plaintext (decrypted) DEK. But to
+    decrypt the DEK, you need the private key associated with the included public key, which
+    you've hopefully kept in a safe, trusted location.
+
+    Note that the supplied file names are load-bearing - ex, changing to `data.key` to `foo.key`
+    risks breaking assumptions that the decryption side will make on the other end!
+    """
+
+    # Generate a new DEK (data encryption key), and use that DEK to encrypt the JSON being exported.
+    pem = encrypt_with.read()
+    data_encryption_key = Fernet.generate_key()
+    backup_encryptor = Fernet(data_encryption_key)
+    encrypted_json_export = backup_encryptor.encrypt(json.dumps(json_export).encode("utf-8"))
+
+    # Encrypt the newly minted DEK using symmetric public key encryption.
```

Comment:
```suggestion
    # Encrypt the newly minted DEK using asymmetric public key encryption.
```

The DEK is encrypted with asymmetric encryption. Once encrypted with the public key, it can only be decrypted with the private key.